### PR TITLE
New v63003/gtm8680 subtest (Tests GTM-8680 in V63003)

### PR DESCRIPTION
**Pull request by VinayYottaDB**
GT.M LOCK commands perform better with large numbers of locks, and particularly with large numbers of processes acquiring the locks. Previously processes acquiring locks could encounter significant slowdown and lock timeouts as the number of locks and competing processes increased. This change requires additional memory per lock slot, so administrators should monitor lock slots (LKE SHOW) to determine if they need to increase lock space needs. The GDE -LOCK_SPACE segment qualifier and MUPIP SET -LOCK_SPACE qualifier accept a maximum value of 262144 pages. Previously the maximum value was 65536 pages. LKE SHOW includes a LOCKSPACEINFO message in its output for regions with a BG or MM access method. This message provides additional information on the use of LOCK space. Previously LKE only issued this message when the lock space was exhausted. (GTM-8680)

### CODE REVIEW

File: v63003/instream.csh [hunk 1]
```diff
@@ -47,6 +47,7 @@
 # gtm8801	    [vinay] Tests ^%YGBLSTAT works on a cmake build
 # gtm8842	    [vinay] Tests TRIGGER_MOD appropriately restricts ZBREAK and ZSTEP
 # gtm8858	    [vinay] Demonstrates the improved available information in cases of apparent database integrity issues
+# gtm8680	    [vinay] Tests YDB does not slow down significantly when holding a large number of locks
```
Comment (nars1) [inline v63003/instream.csh]: when holding a large number of locks -> with large # of locks and/or processes

File: v63003/inref/gtm8680.m [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+;	of its copyright holder(s), and is made available	;
+;	under a license.  If you do not know the terms of	;
+;	the license, please stop and do not read further.	;
+;								;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;
+manylocks
+        set ^stop=0
+        set jmaxwait=0
+	if $ZTRNLNM("proc")  do ^job("mlockschild^gtm8680",1,"""""")
```
Comment (nars1) [inline v63003/inref/gtm8680.m]: Why is the tab indent at the start of the line different compared to surrounding lines?

File: v63003/inref/gtm8680.m [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        . set cur=$h,diff=$$^difftime(cur,prev)  do
+        . if diff>0 do  set prev=cur,previ=i,t=$$^difftime(cur,start)
+	. . if t=2 set maxit=i-previ,minit=i-previ
+	. . if t>=2 do
+	. . . if i-previ>maxit set maxit=i-previ
+	. . . if i-previ<minit set minit=i-previ
+	write "# Max Iterations in a second=",maxit,!
+	write "# Min Iterations in a second=",minit,!
+	set delta=maxit-minit
+	set avg=(maxit+minit)/2
```
Comment (nars1) [inline v63003/inref/gtm8680.m]: Can lines 33 and 34 be simplified to instead say the following (i.e. max should be at most 33% more than min).

if (maxit<(minit*1.33))

File: v63003/outref/gtm8680.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Running test with 1 process
```
Comment (nars1) [inline v63003/outref/gtm8680.txt]: 1 process -> 1 process that creates lots of lock names

File: v63003/outref/gtm8680.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+%YDB-I-LOCKSPACEINFO, Region: DEFAULT: processes on queue: 0/240000; LOCK slots in use: 100001/181132; name space not full
+
+# Running test with 100 processes, each holding 1000 locks
+# Locking ^a(1)-^a(100000) in child processes, attempting to lock ^a(i) in parent
+##TEST_AWK# Max Iterations in a second=[0-9]*
+##TEST_AWK# Min Iterations in a second=[0-9]*
+# No Significant Slowdown Experienced
+##TEST_AWK%YDB-I-LOCKSPACEINFO, Region: DEFAULT: processes on queue: 0/240000; LOCK slots in use: [0-9]*/181132; name space not full
+
+Database file ##TEST_PATH##/mumps.dat now has lock space 65536 pages
```
Comment (nars1) [inline v63003/outref/gtm8680.txt]: Add a comment here that we want to test maximum lock space supported is now bumped from 65536 to 262144

File: v63003/u_inref/gtm8680.csh [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+# All rights reserved.						#
+#								#
+#	This source code contains the intellectual property	#
+#	of its copyright holder(s), and is made available	#
+#	under a license.  If you do not know the terms of	#
+#	the license, please stop and do not read further.	#
+#								#
+#################################################################
+#
+# Tests YDB does not experience significant slowdown with large number locks
```
Comment (nars1) [inline v63003/u_inref/gtm8680.csh]: locks -> of locks and/or processes

File: v63003/u_inref/gtm8680.csh [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+#
+$gtm_tst/com/dbcreate.csh mumps 1 -lock_space=60000>& a.out
```
Comment (nars1) [inline v63003/u_inref/gtm8680.csh]: dbcreate and dbcheck status needs to be checked.

Comment (nars1) [inline v63003/u_inref/gtm8680.csh]: Add a comment that the lockspace of 60000 is needed to ensure we never call the garbage collector (which we can if we have a small lock space) as that will clean up the unused lock names from shared memory thereby speeding up future lock commands even in the version that does not have the GTM-8680 fix.

Comment (nars1) [reply v63003/u_inref/gtm8680.csh]: This comment does not seem to have been incorporated.

Comment (nars1) [reply v63003/u_inref/gtm8680.csh]: This comment does not seem to have been incorporated.

File: v63003/inref/gtm8859.m [hunk 8]
```diff
@@ -0,0 +1,13 @@
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
```
Comment (nars1) [inline v63003/inref/gtm8859.m]: What is the purpose of this module?

File: v63003/u_inref/gtm8859.csh [hunk 9]
```diff
@@ -0,0 +1,15 @@
+#!/usr/local/bin/tcsh -f
```
Comment (nars1) [inline v63003/u_inref/gtm8859.csh]: Same question with this.

File: v63003/u_inref/gtm8680.csh (hunk 7 above)
Comment (nars1) [reply v63003/u_inref/gtm8680.csh]: This comment still does not seem to have been incorporated.
